stdlib/universe/join.go

```diff
@@ -916,7 +918,20 @@ func (c *MergeJoinCache) join(left, right *execute.ColListTableBuilder) (flux.Ta
 						if !c.on[newColumn.Label] {
 							_ = builder.AppendValue(newColumnIdx, columnVal)
 						}
+						trackIdx[newColumnIdx] = struct{}{}
 					})
+
+					// append nil for the missing column in the result table
+					for _, idx := range c.colIndex {
+						if _, ok := trackIdx[idx]; !ok {
+							_ = builder.AppendNil(idx)
+						}
+					}
+
+					// clear map
+					for k := range trackIdx {
+						delete(trackIdx, k)
+					}
```

Comment:
Do you know if the set of columns that need to be padded will be the same for every iteration of the innermost loop? I'm wondering if we could build `trackIdx` before this loop:
https://github.com/influxdata/flux/pull/3895/files#diff-362f933cc9740993880d5db64b38c8776a74dd18d13a458c91801416b1a10f08L890
And then clear it afterwards.